LanguageFeatures/Patterns/pattern_variable_sets_A01_t02.dart

```diff
@@ -24,9 +24,9 @@ String test1(Object? x) {
   switch (x) {
     case int v1 || [int v1] && [_] when v1 == 1:
       return "logical-and";
-    case final int? v2 || final int? v2? when v2 == 2:
+    case final int? v2? || final int? v2? when v2 == 2:
```

Comment:
Aren't we going to get the same dead code / unnecessary `?` warning here for the second `v2?`?

Reply:
Similarly for several later cases.

Reply:
Moved static warnings tests to another PR.

Reply:
OK!